0000-deb-repo-additional-headers.md

```diff
@@ -0,0 +1,107 @@
+- Feature Name: deb_repo_additional_fields
+- Start Date: 12-06-2019
+- RFC PR: (leave this empty)
+
+# Summary
+
+Support additional packages fields in Debian repos.
+
+# Motivation
+
+- Why are we doing this?
+- What use cases does it support?
+- What is the expected outcome?
+
+Uyuni doesn't expose all the metadata coming from the upstream repos. Additional package metadata like `Multi-Arch` is needed by `apt` in some cases to avoid installing a package over and over again.
+Uyuni must be able to create Debian repositories with the correct metadata.
+
+# Detailed design
+
+## Repo syncing
+
+When `spacewalk-repo-sync` parses the `Package.xz|gz` file it must take into account any additional fields.
+Currently it takes into account only theses fields: `Package`, `Architecture`, `Version`, `Filename`, `SHA256`, `SHA1`, `MD5Sum`.
+Any other fields should be saved in the database.
+
+E.g. `Packages` entry:
+```
+Package: apache2-data
+Architecture: all
+Version: 2.4.29-1ubuntu4.6
+Multi-Arch: foreign
+Priority: optional
+Section: httpd
+Source: apache2
+Origin: Ubuntu
+Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
+Original-Maintainer: Debian Apache Maintainers <debian-apache@lists.debian.org>
+Bugs: https://bugs.launchpad.net/ubuntu/+filebug
+Installed-Size: 870
+Filename: pool/main/a/apache2/apache2-data_2.4.29-1ubuntu4.6_all.deb
+Size: 159984
+MD5sum: 7d4e4ad49eb6c5014d5453721573e637
+SHA1: 1824d3d176e7db8010216abc2f0962ac2fe5531f
+SHA256: 594070814770a028a9eb3924d02b22fcb818d83b311710aaac304a9064479292
+Homepage: http://httpd.apache.org/
+Description: Apache HTTP Server (common files)
+Task: lamp-server
+Description-md5: 9f2fab36019a61312dec627d1cd80365
+Supported: 5y
+```
+
+The database schema needs to be extended with a new table to store the additional fields not covered by the existing `rhnPackage*` tables:
+```
+rhnPackageExtraTags
+====================
+id          number not null
+name        string no null
+value       string not null
+created     timestamp
+modified    timestamp
+
+primary key (id)
+```
```

Comment:
General question: should be continue with the maximal normalized DB schema?

I think we will will store a lot of data in this tabel and for the "key" (name) we will have tons of duplicates.
So we might want to have an extra trable `rhnPackageExtraTagsKeys` which just define `id` and `name` and
use `key_id` instead of `name` here in rhnPackageExtraTags.

Could save a lot of space

Reply:
Agree. Will add a new table `rhnPackageExtraTagKey`.